### PR TITLE
[ROCm] Turn off SwapConvOperands on ROCm gpu

### PR DESCRIPTION
**Pull request by jayfurmanek**
The following upstream commit turns on SwapConvOperands on gpu, and this does not yet work on the ROCm platform

tensorflow@43abc0f

Unit Test Failures
```
//tensorflow/compiler/xla/service/gpu/tests:swap_conv_operands_test      FAILED in 3 out of 3 in 18.6s
//tensorflow/compiler/xla/tests:convolution_test_gpu                     FAILED in 3 out of 3 in 67.8s
//tensorflow/compiler/xla/tests:convolution_test_gpu_alternative_layout_gpu FAILED in 3 out of 3 in 63.1s
//tensorflow/compiler/xla/tests:convolution_variants_test_gpu            FAILED in 3 out of 3 in 11.5s
```

This commit
* disables the feature on ROCm to get the unit tests working again
* adds a `no_rocm` tag to the new unit test that was added to specifically test the SwapConvOperands feature

### CODE REVIEW

**Comment by jayfurmanek** [comment]
@deven-amd  FYI

File: tensorflow/compiler/xla/service/gpu/gpu_compiler.cc [hunk 1]
```diff
@@ -444,6 +444,9 @@ Status GpuCompiler::OptimizeHloModule(
       // bitcast(bitcast) with one bitcast. This leads to having to
       // linearize and then delinearize the index.
       options.set_replace_transpose_with_bitcast(false);
+#if TENSORFLOW_USE_ROCM
```
Comment (cheshire) [inline tensorflow/compiler/xla/service/gpu/gpu_compiler.cc:447]: Could we make it a runtime check somehow?

Comment (jayfurmanek) [reply tensorflow/compiler/xla/service/gpu/gpu_compiler.cc]: OK - checked the platform string instead. Is that better? :)

**Comment by jayfurmanek** [comment]
The AMD ROCm fail here to be fixed in: https://github.com/tensorflow/tensorflow/pull/54227

**Review by cheshire — APPROVED** [review]

**Pull request merged**